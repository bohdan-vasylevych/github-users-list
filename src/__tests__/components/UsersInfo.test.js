import React from 'react';
import rendered from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import UserInfo from '../../components/UsersInfo';

configure({adapter: new Adapter()});

const defaultProps = {
    avatar_url: 'avatar_url',
    name: 'name',
    followers: 'followers',
    following: 'following',
    created_at: 'created_at',
    company: 'company',
    email: 'email',
    location: 'location',
    blog: 'blog',
    bio: 'bio'
};

describe('User component', () => {

    test('matches the snapshot', () => {
        const tree = rendered.create(<Router><UserInfo {...defaultProps}/></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
