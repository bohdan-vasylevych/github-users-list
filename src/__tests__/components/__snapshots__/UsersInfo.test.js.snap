// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component matches the snapshot 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    name
  </div>
  <div
    className="panel-body"
  >
    <div
      className="col-md-6"
    >
      <img
        alt="name"
        src="avatar_url"
      />
    </div>
    <div
      className="col-md-6"
    >
      <ul
        className="list-group"
      >
        <li
          className="list-group-item"
        >
          followers: 
          followers
        </li>
        <li
          className="list-group-item"
        >
          following: 
          following
        </li>
        <li
          className="list-group-item"
        >
          created_at: 
          created_at
        </li>
        <li
          className="list-group-item"
        >
          company: 
          company
        </li>
        <li
          className="list-group-item"
        >
          email: 
          email
        </li>
        <li
          className="list-group-item"
        >
          location: 
          location
        </li>
        <li
          className="list-group-item"
        >
          blog: 
          blog
        </li>
        <li
          className="list-group-item"
        >
          bio: 
          bio
        </li>
      </ul>
    </div>
  </div>
</div>
`;
